### PR TITLE
fix: `read_int64_le()` cannot read a large integer

### PR DESCRIPTION
**Pull request by shoamano83**
Fixes https://github.com/smartdevicelink/bson_c_lib/issues/19

This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Testing Plan
Re-run the unit tests.

### Summary
This PR updates the local variable inside `read_int64_le()` to support 64-bit integer.

##### Breaking Changes
* None

##### Enhancements
* Adding unit tests for `read_int64_le()` functions

##### Bug Fixes
* `read_int64_le()` cannot read a large integer

### Tasks Remaining:
- None

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by shoamano83** [comment]
Note: I used libcheck for unit testing since I see a test code `src/emhashmap/tests/tests.c` which looks like using the library. I'm OK to move to another unit testing framework (Google test for example) though.

**Comment by shoamano83** [comment]
Hi @jacobkeeler , could you kindly review? Thanks.

File: Makefile.am [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ACLOCAL_AMFLAGS = -I m4
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = src
+SUBDIRS = src test
```
Comment (jacobkeeler) [inline Makefile.am]: See https://github.com/smartdevicelink/bson_c_lib/pull/22/files#r245103369

Comment (shoamano83) [reply Makefile.am]: OK, I added `--with-tests` option so that `test` subdirectory is disabled by default.

**Comment by shoamano83** [comment]
Good point. Added documentation in commit 88f255f545ab5ade263b6cb2c6f68be83d0b401d. Thanks.

**Pull request merged**